docs/sync-overview.md

```diff
@@ -0,0 +1,57 @@
+# Sync Overview
+
+This document provides a high-level overview of how syncing works.  **Note**: each component has its own quirks and will handle sync slightly differently than the general process described here.
+
+## General flow and architecture
+
+- **Crates involved**:
+  - The `sync15` and `support/sync15-traits` handle the general syncing logic and define the `SyncEngine` trait
+  - `SyncEngine` is implemented by each component crate (login, places, autofill, etc).
+  - `sync_manager` manages the overall syncing process.
+- **High level sync flow**:
+  - Sync is initiated by the application that embeds application-services.
+  - The application calls `SyncManager.sync()` to start the sync process.
+  - `SyncManager` creates `SyncEngine` instances to sync the individual components
+
+### Sync manager
+
+[`SyncManager`](https://github.com/mozilla/application-services/blob/main/components/sync_manager/src/manager.rs) is responsible for performing the high-level parts of the sync process:
+  - The consumer code calls it's `sync()` function to start the sync, passing
+    in a `SyncParams` object in, which describes what should be synced.
+  - `SyncManager` checks if we are currently in a backoff period and should wait before contacting the server again.
+  - `SyncManager` creates invididual `SyncEngine` instances for each component that should be synced.  We currently use 2 different methods for this:
+    - The older method is for the `SyncManager` to store a weakref to the `Store` for each component and use that to create a `SyncEngine` (tabs and places)
```

Comment:
I would love to see some context of what a `Store` is! Maybe a link to somewhere?